src/python/lib/mesos/http.py

```diff
@@ -70,101 +71,210 @@ class Resource(object):
     request, and auth.
     """
 
-    def __init__(self, url, headers=None, timeout_secs=None, auth=None):
+    SUCCESS_CODES = frozenset(xrange(200, 300))
+
+    def __init__(self,
+                 url,
+                 base_headers=BASE_HEADERS,
```

Comment:
using a mutable objects as a default argument can cause unexpected behavior. cf. https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html

this is why `None` was passed in the original code.